api/handlers.go

```diff
@@ -15,43 +18,95 @@ type Sessions interface {
 	New(email string) (*session.Session, error)
 }
 
+// Cache interface for storing and retrieving sessions
+type Cache interface {
+	Set(s *session.Session) error
+	GetByID(ID string) (*session.Session, error)
+}
+
+// GetVarsFunc is a helper function that returns a map of request variables and parameters
+type GetVarsFunc func(r *http.Request) map[string]string
+
 // CreateSessionHandlerFunc returns a function that generates a session. Method = "POST"
-func CreateSessionHandlerFunc(sessions Sessions) http.HandlerFunc {
+func CreateSessionHandlerFunc(sessions Sessions, cache Cache) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
 
 		var c session.NewSessionDetails
 		err := json.NewDecoder(r.Body).Decode(&c)
 		if err != nil {
-			log.Event(ctx, "failed to unmarshal request body", log.Error(err), log.ERROR)
-			http.Error(w, "Failed to unmarshal request body", http.StatusBadRequest)
+			writeErrorResponse(ctx, w, "failed to unmarshal request body", err, http.StatusBadRequest)
 			return
 		}
 
 		if len(c.Email) == 0 {
-			log.Event(ctx, "missing email field in json", log.ERROR)
-			http.Error(w, "Missing email field in json", http.StatusBadRequest)
+			writeErrorResponse(ctx, w, "missing email field in json", err, http.StatusBadRequest)
 			return
 		}
 
 		sess, err := sessions.New(c.Email)
 		if err != nil {
-			log.Event(ctx, "failed to create session", log.Error(err), log.ERROR)
-			http.Error(w, "Failed to create session", http.StatusInternalServerError)
+			writeErrorResponse(ctx, w, "failed to create session", err, http.StatusInternalServerError)
 			return
 		}
 
-		log.Event(ctx, "session created", log.Data{"session": sess}, log.INFO)
+		s := &session.Session{
+			ID:    sess.ID,
+			Email: sess.Email,
+			Start: sess.Start,
+		}
+
+		if err := cache.Set(s); err != nil {
+			writeErrorResponse(ctx, w, "unable to add session to cache", err, http.StatusInternalServerError)
+			return
+		}
 
-		sessJson, err := sess.MarshalJSON()
+		sessJSON, err := sess.MarshalJSON()
 		if err != nil {
-			log.Event(ctx, "failed to marshal session", log.Error(err), log.ERROR)
-			http.Error(w, "Failed to marshal session", http.StatusInternalServerError)
+			writeErrorResponse(ctx, w, "failed to marshal session", err, http.StatusInternalServerError)
 			return
 		}
 
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(http.StatusCreated)
-		w.Write(sessJson)
+		w.Write(sessJSON)
+	}
+}
+
+func GetByIDSessionHandlerFunc(cache Cache, getVarsFunc GetVarsFunc) http.HandlerFunc {
```

Comment:
Should have GoDoc if this is an exported function. 
```suggestion
// GetByIDSessionHandlerFunc does this that or the other
func GetByIDSessionHandlerFunc(cache Cache, getVarsFunc GetVarsFunc) http.HandlerFunc {
```